tests/integration-tests/resources/file-cache-storage-cfn.yaml

```diff
@@ -0,0 +1,261 @@
+      VpcId: !Ref VpcId
+
+  CreateFsxFileCacheLambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      Description: !Sub "${AWS::StackName}: custom resource handler to create FSX File Cache."
+      Handler: index.handler
+      MemorySize: 128
+      Role: !GetAtt FsxFileCacheLambdaRole.Arn
+      Runtime: python3.9
+      Timeout: 300
+      TracingConfig:
+        Mode: Active
+      Code:
+        ZipFile: |
+          import time
+          import cfnresponse
+          import boto3
+          import logging
+          import random
+          import string
+          logger = logging.getLogger()
+          logger.setLevel(logging.INFO)
+
+          fsx = boto3.client("fsx")
+
+          def create_physical_resource_id():
+            alnum = string.ascii_uppercase + string.ascii_lowercase + string.digits
+            return ''.join(random.choice(alnum) for _ in range(16))
+
+          def create_file_cache_id(file_cache_path, s3_bucket, subnet_id, securitygroup_id):
+            logger.info(f"Creating Fsx File Cache ...")
+            try:
+              return (
+                fsx.create_file_cache(
+                  FileCacheType="LUSTRE",
+                  FileCacheTypeVersion="2.12",
+                  StorageCapacity=1200,
+                  SubnetIds=[subnet_id],
+                  SecurityGroupIds=[securitygroup_id],
+                  LustreConfiguration={
+                    "PerUnitStorageThroughput": 1000,
+                    "DeploymentType": "CACHE_1",
+                    "MetadataConfiguration": {"StorageCapacity": 2400},
+                  },
+                  DataRepositoryAssociations=[
+                    {
+                      "FileCachePath": file_cache_path,
+                      "DataRepositoryPath": s3_bucket,
+                    },
+                  ],
+                ).get("FileCache")
+              )
+
+            except fsx.exceptions.InternalServerError as e:
+              logger.info(f"Cannot create Fsx FileCache because of InternalServerError.")
+
+
+          def handler(event, context):
+            logger.info(f"Context: {context}")
+            logger.info(f"Event: {event}")
+            logger.info(f"Boto version: {boto3.__version__}")
+
+            file_cache_path =  event['ResourceProperties']['FileCachePath']
+            logger.info(f"file_cache_path: {file_cache_path}")
+
+            s3_bucket = 's3://{}'.format(event['ResourceProperties']['S3BucketName'])
+            logger.info(f"s3_bucket: {s3_bucket}")
+
+            subnet_id = event['ResourceProperties']['SubnetId']
+            logger.info(f"subnet_id: {subnet_id}")
+
+            securitygroup_id = event['ResourceProperties']['SecurityGroupId']
+            logger.info(f"securitygroup_id: {securitygroup_id}")
+
+
+            response_data = {}
+            reason = None
+            response_status = cfnresponse.SUCCESS
+
+            try:
+              if event['RequestType'] == 'Create':
```

Comment:
The custom resource should take care also for the deletion of the resource, not only for the creation.

Reply:
The reason i separated the Creation and Deletion was it helped in debugging. But I see your point in combining the two as both the creation/deletion logic is stand alone anyway.